fiberplane-models/src/providers/auto_suggest.rs

```diff
@@ -14,14 +14,19 @@ use typed_builder::TypedBuilder;
 )]
 #[non_exhaustive]
 pub struct AutoSuggestRequest {
-    /// The query being typed by the user, up to the focus offset.
+    /// The field being typed by the user, up to the focus offset.
     pub query: String,
 
     /// The query type of the provider we're requesting suggestions for.
     pub query_type: String,
 
     /// The field in the query form we're requesting suggestions for.
     pub field: String,
+
+    /// The current serialized request in its entirety.
+    /// It is not guaranteed to be valid, but each provider can use this to
+    /// provide context-aware suggestions.
+    pub full_query: String,
```

Comment:
This field must be optional to avoid a breaking change. (I also don't think it makes sense to make it mandatory semantically.)

I find the name and the doc comment to be somewhat lacking. It's called "full" query, but it may contain partial data. How is that data encoded? Does it contain the `field` and `query` in duplicated form? If so, why?

I think I would go for something like this instead:

```suggestion
    /// If other fields in the provider's query form have already been filled in, this may contain a
    /// form-encoded query string with the data for those fields.
    pub other_field_data: Option<String>,
```

Reply:
It contains the full content of the cell's data, encoded as a query. There's no special filtering on what data it's supposed to have, so it will likely contain the data in existing fields one additional time. Technically this should be called "query", because it contains the query represented by the cell, but the name query is already taken for the "current field input" data, so I thought "full query" was good.